### PR TITLE
Auto updating docs

### PR DESCRIPTION
**Pull request by InFTord**
Make sure you familiarize yourself with our contributing guidelines.

# Summary
Adds auto updating docs via action and GH Pages.

# Details
Adds a new action, that builds and deploys our docs to GH Pages.
Currently WIP, since i need to fix some issues.

# Notes
Should i touch JS files in dsplus/ folder in docs?

### CODE REVIEW

**Comment by InFTord** [comment]
@OoLunar if  you wanna, go check this action and suggest how we can improve it :3

**Comment by InFTord** [comment]
NOTE: deploy is failing because of permissions.
token permissions

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+env:
+  DOTNET_NOLOGO: true
+  DOTNET_CLI_TELEMETRY_OPTOUT: true
+
+permissions:
+  contents: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (OoLunar) [inline .github/workflows/docs.yml:18]: Self hosted

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
+name: .NET Docs
+
+on:
+  push:
```
Comment (OoLunar) [inline .github/workflows/docs.yml:4]: On release

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      with:
+        dotnet-version: 7.0.x
+
+    - name: Install DocFX
+      run: dotnet tool update -g docfx
+
+    - name: Build documentation
+      run: docfx ./docs/docfx.json 
+
+
```
Comment (OoLunar) [inline .github/workflows/docs.yml:33]: Double newline

File: .github/workflows/docs.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      pages: write      # to deploy to Pages
+      id-token: write   # to verify the deployment originates from an appropriate source
+
+    # Deploy to the github-pages environment
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+
+    # Specify runner + deployment step
+    runs-on: ubuntu-latest
```
Comment (OoLunar) [inline .github/workflows/docs.yml:55]: Self hosted

File: .github/workflows/docs.yml (hunk 1 above)
Comment (InFTord) [reply .github/workflows/docs.yml]: fixed

File: .github/workflows/docs.yml (hunk 2 above)
Comment (InFTord) [reply .github/workflows/docs.yml]: fixed

File: .github/workflows/docs.yml (hunk 3 above)
Comment (InFTord) [reply .github/workflows/docs.yml]: fixed

File: .github/workflows/docs.yml (hunk 4 above)
Comment (InFTord) [reply .github/workflows/docs.yml]: fixed

File: docs/templates/darkfx/partials/head.tmpl.partial [hunk 5]
```diff
@@ -1,4 +1,4 @@
-{{!Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.}}
+{{!Copyright (c) Oscar Vasquez. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.}}
```
Comment (Naamloos) [inline docs/templates/darkfx/partials/head.tmpl.partial:1]: huh

File: .github/workflows/docs.yml [hunk 6]
```diff
@@ -0,0 +1,56 @@
+name: .NET Docs
+
+on:
+  release:
+    types: [published]
```
[the remark this answers is not in the document]
Comment (InFTord) [reply .github/workflows/docs.yml:5]: we can! 
filter-paths

[the remark this answers is not in the document]
Comment (InFTord) [reply .github/workflows/docs.yml:5]: one sec, will actually test it out

[the remark this answers is not in the document]
Comment (akiraveliara) [reply .github/workflows/docs.yml:5]: github has this feature built-in, something like this: ~~~yml
on:
  push:
    branches:
      - main
    paths:
      - docs/**
~~~

i was moreso wondering whether you can combine this with on-release builds

[the remark this answers is not in the document]
Comment (InFTord) [reply .github/workflows/docs.yml:5]: cursed formatting moment

[the remark this answers is not in the document]
Comment (InFTord) [reply .github/workflows/docs.yml:5]: anywho, ty for your review

**Review by akiraveliara — APPROVED** [review]

**Comment by InFTord** [comment]
finally, im actually done with this PR.
but something to note: there is extra configuration required to make release deploying work.
изображение
environment is automatically created by workflow

**Review by Naamloos — APPROVED** [review]

**Pull request merged**